pkg/controller/workload/workload_processor_test.go

```diff
@@ -26,9 +26,100 @@ import (
 	"kmesh.net/kmesh/api/v2/workloadapi"
 	"kmesh.net/kmesh/daemon/options"
 	"kmesh.net/kmesh/pkg/constants"
+	"kmesh.net/kmesh/pkg/controller/workload/bpfcache"
+	"kmesh.net/kmesh/pkg/nets"
 	"kmesh.net/kmesh/pkg/utils/test"
 )
 
+func Test_handleWorkload(t *testing.T) {
+	workloadMap := bpfcache.NewFakeWorkloadMap(t)
+	defer bpfcache.CleanupFakeWorkloadMap(workloadMap)
+
+	p := newProcessor(workloadMap)
+
+	// 1. handle workload with service, but service not handled yet
+	// In this case, only frontend map and backend map should be updated.
+	wl := createTestWorkloadWithService()
+	_ = p.handleDataWithService(createTestWorkloadWithService())
+	var (
+		ek bpfcache.EndpointKey
+		ev bpfcache.EndpointValue
+	)
+
+	workloadID := checkFrontEndMap(t, wl.Addresses[0], p)
+	checkBackendMap(t, p, workloadID, wl)
+
+	epKeys := p.bpf.EndpointIterFindKey(workloadID)
+	assert.Equal(t, len(epKeys), 0)
+	for svcName := range wl.Services {
+		endpoints := p.endpointsByService[svcName]
+		assert.Len(t, endpoints, 1)
+		if _, ok := endpoints[wl.Uid]; ok {
+			assert.True(t, ok)
+		}
+	}
+
+	// 2. add related service
+	fakeSvc := createFakeService("testsvc", "10.240.10.1", "10.240.10.2")
+	_ = p.handleService(fakeSvc)
+
+	// 2.1 check front end map contains service
+	svcID := checkFrontEndMap(t, fakeSvc.Addresses[0].Address, p)
+
+	// 2.2 check service map contains service
+	checkServiceMap(t, p, svcID, fakeSvc, 1)
+
+	// 2.3 check endpoint map now contains the workloads
+	ek.BackendIndex = 1
+	ek.ServiceId = svcID
+	err := p.bpf.EndpointLookup(&ek, &ev)
+	assert.NoError(t, err)
+	assert.Equal(t, ev.BackendUid, workloadID)
+
+	// 3. add another workload with service
+	workload2 := createFakeWorkload("1.2.3.5")
+	_ = p.handleDataWithService(workload2)
+
+	// 3.1 check endpoint map now contains the new workloads
+	workload2ID := checkFrontEndMap(t, workload2.Addresses[0], p)
+	ek.BackendIndex = 2
+	ek.ServiceId = svcID
+	err = p.bpf.EndpointLookup(&ek, &ev)
+	assert.NoError(t, err)
+	assert.Equal(t, ev.BackendUid, workload2ID)
+
+	// 3.2 check service map contains service
+	checkServiceMap(t, p, svcID, fakeSvc, 2)
+}
+
+func checkServiceMap(t *testing.T, p *Processor, svcId uint32, fakeSvc *workloadapi.Service, endpointCountchan uint32) {
```

Comment:
Why is it named `endpointCountchan`? It sounds like a channel

Reply:
Ah, it is supplemented by  copilot. Will remove